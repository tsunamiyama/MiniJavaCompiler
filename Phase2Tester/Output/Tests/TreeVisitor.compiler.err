java.lang.NullPointerException
	at vaporVisitor.visit(vaporVisitor.java:145)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.AllocationExpression.accept(AllocationExpression.java:38)
	at syntaxtree.NodeChoice.accept(NodeChoice.java:24)
	at vaporVisitor.visit(vaporVisitor.java:129)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.PrimaryExpression.accept(PrimaryExpression.java:30)
	at syntaxtree.NodeChoice.accept(NodeChoice.java:24)
	at vaporVisitor.visit(vaporVisitor.java:114)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.Expression.accept(Expression.java:30)
	at vaporVisitor.visit(vaporVisitor.java:94)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.AssignmentStatement.accept(AssignmentStatement.java:38)
	at syntaxtree.NodeChoice.accept(NodeChoice.java:24)
	at vaporVisitor.visit(vaporVisitor.java:86)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.Statement.accept(Statement.java:27)
	at visitor.GJDepthFirst.visit(GJDepthFirst.java:32)
	at syntaxtree.NodeListOptional.accept(NodeListOptional.java:34)
	at vaporVisitor.visit(vaporVisitor.java:68)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.MethodDeclaration.accept(MethodDeclaration.java:74)
	at visitor.GJDepthFirst.visit(GJDepthFirst.java:32)
	at syntaxtree.NodeListOptional.accept(NodeListOptional.java:34)
	at vaporVisitor.visit(vaporVisitor.java:57)
	at vaporVisitor.visit(vaporVisitor.java:8)
	at syntaxtree.ClassDeclaration.accept(ClassDeclaration.java:46)
	at syntaxtree.NodeChoice.accept(NodeChoice.java:24)
	at visitor.GJDepthFirst.visit(GJDepthFirst.java:126)
	at syntaxtree.TypeDeclaration.accept(TypeDeclaration.java:23)
	at visitor.GJDepthFirst.visit(GJDepthFirst.java:32)
	at syntaxtree.NodeListOptional.accept(NodeListOptional.java:34)
	at visitor.GJDepthFirst.visit(GJDepthFirst.java:72)
	at syntaxtree.Goal.accept(Goal.java:34)
	at J2V.main(J2V.java:20)
