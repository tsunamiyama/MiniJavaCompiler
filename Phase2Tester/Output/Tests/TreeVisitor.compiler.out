const vmt_TV
	:TV.Start

const vmt_Visitor
	:Visitor.visit

const vmt_Tree
	:Tree.Init
	:Tree.SetRight
	:Tree.SetLeft
	:Tree.GetRight
	:Tree.GetLeft
	:Tree.GetKey
	:Tree.SetKey
	:Tree.GetHas_Right
	:Tree.GetHas_Left
	:Tree.SetHas_Left
	:Tree.SetHas_Right
	:Tree.Compare
	:Tree.Insert
	:Tree.Delete
	:Tree.Remove
	:Tree.RemoveRight
	:Tree.RemoveLeft
	:Tree.Search
	:Tree.Print
	:Tree.RecPrint
	:Tree.accept

func Main()
	t.0 = HeapAllocZ(4)
	[t.0] = :vmt_TV
	t.1 = t.0
	if t.1 goto :null1
		Error("null pointer")
	null1:
	t.2 = [t.1]
	t.2 = [t.2+0]
	t.3 = call t.2(t.1)
	t.4 = t.3
	PrintIntS(t.4)
	ret

func TV.Start(this)
	t.0 = HeapAllocZ(28)
	[t.0] = :vmt_Tree
	t.1 = t.0
	t.2 = t.1
	root = t.2
	t.3 = root
	t.4 = 16
	t.5 = t.4
	if t.3 goto :null2
		Error("null pointer")
	null2:
	t.6 = [t.3]
	t.6 = [t.6+0]
	t.7 = call t.6(t.3 t.5)
	t.8 = t.7
	ntb = t.8
	t.9 = root
	if t.9 goto :null3
		Error("null pointer")
	null3:
	t.10 = [t.9]
	t.10 = [t.10+72]
	t.11 = call t.10(t.9)
	t.12 = t.11
	ntb = t.12
	t.13 = 100000000
	t.14 = t.13
	PrintIntS(t.14)
	t.15 = root
	t.16 = 8
	t.17 = t.16
	if t.15 goto :null4
		Error("null pointer")
	null4:
	t.18 = [t.15]
	t.18 = [t.18+48]
	t.19 = call t.18(t.15 t.17)
	t.20 = t.19
	ntb = t.20
	t.21 = root
	t.22 = 24
	t.23 = t.22
	if t.21 goto :null5
		Error("null pointer")
	null5:
	t.24 = [t.21]
	t.24 = [t.24+48]
	t.25 = call t.24(t.21 t.23)
	t.26 = t.25
	ntb = t.26
	t.27 = root
	t.28 = 4
	t.29 = t.28
	if t.27 goto :null6
		Error("null pointer")
	null6:
	t.30 = [t.27]
	t.30 = [t.30+48]
	t.31 = call t.30(t.27 t.29)
	t.32 = t.31
	ntb = t.32
	t.33 = root
	t.34 = 12
	t.35 = t.34
	if t.33 goto :null7
		Error("null pointer")
	null7:
	t.36 = [t.33]
	t.36 = [t.36+48]
	t.37 = call t.36(t.33 t.35)
	t.38 = t.37
	ntb = t.38
	t.39 = root
	t.40 = 20
	t.41 = t.40
	if t.39 goto :null8
		Error("null pointer")
	null8:
	t.42 = [t.39]
	t.42 = [t.42+48]
	t.43 = call t.42(t.39 t.41)
	t.44 = t.43
	ntb = t.44
	t.45 = root
	t.46 = 28
	t.47 = t.46
	if t.45 goto :null9
		Error("null pointer")
	null9:
	t.48 = [t.45]
	t.48 = [t.48+48]
	t.49 = call t.48(t.45 t.47)
	t.50 = t.49
	ntb = t.50
	t.51 = root
	t.52 = 14
	t.53 = t.52
	if t.51 goto :null10
		Error("null pointer")
	null10:
	t.54 = [t.51]
	t.54 = [t.54+48]
	t.55 = call t.54(t.51 t.53)
	t.56 = t.55
	ntb = t.56
	t.57 = root
	if t.57 goto :null11
		Error("null pointer")
	null11:
	t.58 = [t.57]
	t.58 = [t.58+72]
	t.59 = call t.58(t.57)
	t.60 = t.59
	ntb = t.60
	t.61 = 100000000
	t.62 = t.61
	PrintIntS(t.62)
